src/ex15_js-oop/task-01.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+function Candy(brand, weight) {
+  this.brand = brand;
+  this.weight = weight;
+}
+function SugarCandy(brand, weight, type) {
+  Candy.call(this, brand, weight);
+  this.type = type;
+}
+function ChocolateCandy(brand, weight, type) {
+  Candy.call(this, brand, weight);
+  this.type = type;
+}
+SugarCandy.prototype = Object.create(Candy.prototype);
+SugarCandy.prototype.constructor = SugarCandy;
+ChocolateCandy.prototype = Object.create(Candy.prototype);
+ChocolateCandy.prototype.constructor = ChocolateCandy;
+function Gift() {
+  this.candies = [];
+}
+Gift.prototype.addCandy = function(candy, qty) {
+  for (let i = 0; i < qty; i++) {
+    this.candies.push(candy);
+  }
+  return this;
+}
+Gift.prototype.getTotalWeight = function() {
+  let totalWeight = 0;
+  this.candies.forEach(function(candy) { totalWeight += candy.weight });
+  return totalWeight.toFixed(3);
+}
+Gift.prototype.sortCandies = function(byProp, selectedType) {
+  switch(byProp) {
+    case 'brand':
+    default:
+      return this.candies.sort((candy1, candy2) => {
+        if (candy1.brand.toLowerCase() < candy2.brand.toLowerCase()) {
+          return -1;
+        }
+        if (candy1.brand.toLowerCase() > candy2.brand.toLowerCase()) {
+          return 1;
+        }
+        return 0;
+      })
+    break;
+    case 'type':
+      return this.candies.filter(candy => candy.type === selectedType);
+    break;
+  }
+}
+Gift.prototype.findCandy = function(byBrand) {
+  return this.candies.find(candy => candy.brand.toLowerCase() === byBrand.toLowerCase());
```

Comment:
why do u use `by` before prop name?

Reply:
I thought "find candy 'by brand'" would sound somehow more declarative. But probably that was a bad idea.

Reply:
yep, cause of u use it out from this func
argument's name need to be more neutral

Reply:
Renamed this too